tests/filecheck/dialects/riscv/riscv_ops.mlir

```diff
@@ -397,6 +397,6 @@
 // CHECK-GENERIC-NEXT:     %fmv_w_x = "riscv.fmv.w.x"(%0) : (!riscv.reg<>) -> !riscv.freg<>
 // CHECK-GENERIC-NEXT:     %flw = "riscv.flw"(%0) {"immediate" = 1 : si12} : (!riscv.reg<>) -> !riscv.freg<>
 // CHECK-GENERIC-NEXT:     "riscv.fsw"(%0, %f0) {"immediate" = 1 : si12} : (!riscv.reg<>, !riscv.freg<>) -> ()
-// CHECK-GENERIC-NEXT:     "riscv.ret"() : () -> ()
-// CHECK-GENERIC-NEXT:   }) {"label" = #riscv.label<"main">} : () -> ()
+// CHECK-GENERIC-NEXT:     "riscv_func.return"() : () -> ()
+// CHECK-GENERIC-NEXT:   }) {"sym_name" = "main", "function_type" = () -> ()} : () -> ()
```

Comment:
Just curious why we need `"function_type" ` attributes in `FuncOp`.

Reply:
We copied this over from `func.func`, my understanding is that it's useful for external function declarations, among other things. At one point we might want to adopt the FunctionOpInterface from MLIR and it's a requirement there AFAIK. https://mlir.llvm.org/docs/Dialects/Func/#funcfunc-funcfuncop